### DISCUSSION TITLE
Ways to get named type from Static? Issues?

### DISCUSSION THREAD

**Discussion by ghost** [sinclairzx81/typebox#911] [Q&A] [answered]
I have found one way to get a named type [1] out of `Static<typeof MySchema>` and that is to extend it onto an interface.

```ts
export const MySchema = Type.Object({ /* ...fields... */ });

// export type MySchema = Static<typeof MySchema>; // standard method

export interface MySchema extends Static<typeof MySchema> {} // produces named type
```

This seems like a pretty standard TypeScript technique and I don't expect any issues to arise from it. However,

- Should I expect any issues?
- And, are there any other ways of doing this?

[1] Without a named type, when I hover over `data` in `(data: MySchema) => {}` I see `data: { ...fields... }`. With a named type, I see `MySchema`.

**Comment by ghost** [comment]
The only other way of doing it that I can think of is to use code generation and stop using `Static`.

I found a couple of instances of it [being used in the wild](https://www.google.com/search?q=%22typebox%22+%22extends+static%22) and I've used interface/extends Static in my own projects, so I shall continue to do so.

**Reply by sinclairzx81** [reply, marked as answer]
@waynebloss Hi, 

Ah, that's a interesting approach. There isn't really a standard / non-standard way to produce named type in TypeBox, but the technique you suggest seems very reasonable to me, I actually quite like it.

In terms of issues, I wouldn't anticipate any, but just to be mindful that `extends Static<T>` is only workable for types that produce structural object types. In terms of other approaches, there's none that I know of (as all conditional mapped types are computed type aliases), but if there is a technique, i would be interested in including that in TypeBox, the following would be ideal.

```typescript
const T = NamedStatic<typeof T> // this would be ideal
```
Maybe one day :)

Hope this helps
S
